### PR TITLE
Image promotion for etcd 3.5.10-0

### PR DESCRIPTION
**Pull request by mzaian**
Based on: https://github.com/kubernetes/kubernetes/pull/121567

```
➜  ~ docker inspect gcr.io/k8s-staging-etcd/etcd:3.5.10-0
[
    {
        "Id": "sha256:a0eed15eed4498c145ef2f1883fcd300d7adbb759df73c901abd5383dda668e7",
        "RepoTags": [
            "gcr.io/k8s-staging-etcd/etcd:3.5.10-0"
        ],
        "RepoDigests": [
            "gcr.io/k8s-staging-etcd/etcd@sha256:22f892d7672adc0b9c86df67792afdb8b2dc08880f49f669eaaa59c47d7908c2"
        ],
```

Required for: https://github.com/kubernetes/kubernetes/pull/121566

### CODE REVIEW

**Comment by pacoxu** [comment]
https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/post-kubernetes-push-image-etcd/1718958005506019328 
failed after https://github.com/kubernetes/kubernetes/pull/121567 was merged.

File: registry.k8s.io/images/k8s-staging-etcd/images.yaml [hunk 1]
```diff
@@ -18,6 +18,7 @@
     "sha256:51eae8381dcb1078289fa7b4f3df2630cdc18d09fb56f8e56b41c40e191d6c83": ["3.5.7-0"]
     "sha256:b3a0da984f9af598234c018faa4f28dd6a6461f5719163a61f10ebf26dbca1e9": ["3.5.8-0"]
     "sha256:e013d0d5e4e25d00c61a7ff839927a1f36479678f11e49502b53a5e0b14f10c3": ["3.5.9-0"]
+    "sha256:": ["3.5.10-0"]
```
Comment (jmhbnz) [inline registry.k8s.io/images/k8s-staging-etcd/images.yaml:21]: Looks like the sha256 has not been added, refer lines above for 3.5.9 and earlier.

**Comment by dims** [comment]
image

/approve
/lgtm

**Review by saschagrunert — APPROVED** [review]

**Pull request merged**